components/permissions/contexts/brave_wallet_permission_context.cc

```diff
@@ -163,9 +164,12 @@ void BraveWalletPermissionContext::RequestPermissions(
   }
 
   auto* web_contents = content::WebContents::FromRenderFrameHost(rfh);
-  // Fail the request came from 3p origin.
-  if (web_contents->GetMainFrame()->GetLastCommittedOrigin() !=
-      rfh->GetLastCommittedOrigin()) {
+  // Fail the request came from a different eTLD+1 origin.
+  if (!rfh->GetParent() &&
```

Comment:
This should be `rfh->GetParent()` if we are looking for iframe.

Reply:
Fixed in the latest